### ISSUE TITLE
On a Mac gettting 'Could not locate 'iTunesMetaData.plist in archived error 

### ISSUE THREAD

**Issue by tommy-banana** [Gamua/Adobe-Runtime-Support#768]
On a Mac getting 'Could not locate 'iTunesMetaData.plist in archive error.

Using AIR SDK 33.1.1.44 and Animate 20.5.1

Publishing to an attached iPhone 8Plus.

Any ideas?

Best Tommy B,

* Which AIR SDK did you use? Hint: always try the latest version from [Adobe Labs](http://labs.adobe.com/downloads/air.html).
* Describe the affected environment: Flash vs. AIR, Android vs. iOS, all devices vs. just specific models.
* Can you reproduce it yourself, or did you get reports from other users? If so, how many users are affected (percentage)?
* If applicable, link to the Starling/Feathers issues that brought you here.

Furthermore, **always** add a link to one or more analog [Adobe Tracker](https://tracker.adobe.com) issues. This repository is not a replacement for Adobe's bug tracker!

## Steps to Reproduce

Tell us how this issue can be reproduced. Seriously: if you want this issue to be fixed, this is the part you **really** need to get nailed.
Give us a minimal example:

* As little code as possible. Get to the root of the problem!
* As few dependencies as possible. Did the problem pop up with Feathers? Try to reproduce it with Starling only.
* Remove any ANEs (except if they are vital for the problem).
* Include all necessary files (e.g. textures).

## Known Workarounds

Just in case others run into this – do you happen to know of a workaround? Otherwise, just remove this section altogether.

**Comment by tommy-banana** [comment]
Should I update Animate to 21.0.4?

**Comment by ajwfrost** [comment]
Hi

There are two lines that the deployment tool outputs as warnings:
```
WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/APPNAME.app/SC_Info/APPNAME.sinf in archive!
```
but these are just warnings, they should not actually affect the functionality.

Does the application appear on the handset? If not we have another problem...

To do this manually, and get the most information out of it, if you go into a terminal window and then do:
```
cd AIRSDK_Folder/lib/aot/bin/iOSBin
./ideviceinstaller -i IPAFILENAME.ipa
```
it should give you a clearer picture. If that doesn't work, please add the debug flag (per the below) and post the output:
```
./ideviceinstaller -d -i IPAFILENAME.ipa
```

thanks

**Comment by tommy-banana** [comment]
WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/ABCInstruments.app/SC_Info/ABCInstruments.sinf in archive!
Copying '/Users/tommyb/Google Drive/All Content Devt/Literacy & Numeracy/Gameification Work/Instrument App/Instruments IPad/IPAD-INSTRUMENTS/bin/ABCInstruments.ipa' to device... DONE.
Installing 'com.abccreativemusic.abcinstruments'
Install: CreatingStagingDirectory (5%)
Install: ExtractingPackage (15%)
Install: InspectingPackage (20%)
Install: TakingInstallLock (20%)
Install: PreflightingApplication (30%)
Install: InstallingEmbeddedProfile (30%)
Install: VerifyingApplication (40%)ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008015: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.2Qv5LE/extracted/Payload/ABCInstruments.app : 0xe8008015 (A valid provisioning profile for this executable was not found.)
tommyb@Thomass2020MBP iOSBin % 

Got this. Will try generating a new provisioning profile. Weird because this one hasn't expired and  hardly changed anything on the ipa.

**Comment by tommy-banana** [comment]
Thanks for the support btw!

**Comment by tommy-banana** [comment]
Made a new provisioning profile - no difference. Maybe I need all new certificates.....

**Comment by ajwfrost** [comment]
That's a strange error code though, especially if you've checked the provisioning profile is right for the certificate and device (and presumably for the application ID as well). What sort of build is it? and are you able to share the common name of your signing certificate?

thanks

**Comment by tommy-banana** [comment]
is it the p12 name you want? . It is an Appstore build. The ipa publishes successfully. I'll see if i can load it on the device using Testflight.

**Comment by tommy-banana** [comment]
It uploaded to AppStore ok and was accepted - which suggests profiles/certificates etc are ok. Will see if it wlll load onto device via testflight.

**Comment by tommy-banana** [comment]
Yes it is working fine viaTestflight.

**Comment by ajwfrost** [comment]
Okay that's odd then that it won't work via the ideviceinstaller tool.. the internet is full of suggestions for what happens when it gives that error, so not entirely sure why this is happening :-(

**Comment by stoff99** [comment]
Hi,
i get exactly the same error:
WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/Plentyco.app/SC_Info/Plentyco.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008001: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.ztmppk/extracted/Payload/Plentyco.app : 0xe8008001 (An unknown error has occurred.)

IDEA: IntelliJ IDEA
iPhone 11 pro iOS 14.4.2

I have the problem with **33.1.1.406** instead of **444**
With Air **33.1.1444** its working but with **33.1.1.406** not.
But for Android i need 33.1.1.406 because 444 has this bug: https://github.com/Gamua/Adobe-Runtime-Support/issues/758
;)

**Comment by tommy-banana** [comment]
Hi I found that if I didnt try to publisj into a connected iOS device ( using Animate and AIR) the ipa would be published and it worked fine, I could upload with Testflight and install it in a device that way.

**Comment by stoff99** [comment]
Hey @tommy-banana,
thx for your answer and help but i rly prefer to develop and debug in the same SDK version as i send to my testers and in the store.
So lets hope that the new AIR release arrives soon @ajwfrost ?

**Comment by ajwfrost** [comment]
@stoff99 yes.. QA found an issue so slightly delayed but targeting Monday now.
Not sure whether it will solve your problem though: these two reports are different:
```
Got error "ApplicationVerificationFailed" with code 0xe8008015: Failed to verify code signature of app : 0xe8008015 (A valid provisioning profile for this executable was not found.)
```
vs
```
Got error "ApplicationVerificationFailed" with code 0xe8008001: Failed to verify code signature of app : 0xe8008001 (An unknown error has occurred.)
```
We're currently looking to work around all these code-signing things for folk using MacOS by just using codesign directly...

**Comment by stoff99** [comment]
@ajwfrost sry, you are right, its a different error code.
Just found this right now: https://forum.starling-framework.org/d/22969-got-error-applicationverificationfailed-with-code-0xe8008001

Looks like you provided a fixed ADT.jar.
But anyway, this is fixed with 444 already, so lets wait for the next build.

Thx for your great work of your team and you!
